### ISSUE TITLE
Deprecated items - candidates for removal in 8.15

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#13538]
The following commands, tactics, flags, command line options, etc are marked as deprecated in the noted versions.  Therefore they are candidates for removal from the code and documentation for 8.14.  Please let me know which of these should retained for 8.14.  I will submit a PR later (perhaps multiple PRs as long the number of PRs is quite small).

EDITED: Items that were commented on that should not be removed yet are shown in strikeout type.

**8.12 and earlier**

```
omega 8.12
   (also remove entire omega chapter)
Stable Omega flag 8.5
Hide Obligations flag 8.12
Bracketing Last Introduction Pattern flag 8.10
convert_concl_no_check 8.11
double induction  (no version)
SearchHead 8.12

-extra-phony and -extra (coq_makefile)
warn_deprecated_instantiate_syntax: generates message for syntax:
   | [ "in" "(" "Type" "of" ident(id) ")" ] ->
   | [ "in" "(" "Value" "of" ident(id) ")" ] ->
"The syntax [autoapply ... using] is deprecated. Use [autoapply ... with] instead." (warning)
```

~~Search (ssr) 8.12~~    REMOVAL POSTPONED TO 8.15

~~Add Setoid, Add Morphism and Declare Morphism (no version)
   (also remove doc section "Deprecated syntax and backward incompatibilities section")
Include Type 8.3
Focus 8.8
Unfocus 8.8~~

**8.13**

Is it too soon to do these, too?

```
Add InjTyp and 10 other commands in micromega (commit date is 5/25/20, so I think 8.13)
universes(monomorphic), universes(noncumulative), universes(notemplate)  attributes
```

~~Instance Generalized Output flag
Existential
Grab Existential Variables~~

**Comment by SkySkimmer** [comment]
>Add Setoid, Add Morphism and Declare Morphism (no version)

IIRC we were thinking about undeprecating these

>Include Type 8.3

Is that a typoed 8.13?

>Instance Generalized Output flag

I'd be happy to see it go but maybe a bit early.

**Comment by jfehrle** [comment]
The doc says `Include Type` is 8.3, but it may be a typo on my part.  I made the change in that line (commit date 2/26/20) but I don't recall any other details.  A lot of the versions are from the doc, BTW.

**Comment by SkySkimmer** [comment]
Looks like it is indeed 8.3, but the removal is stalled https://github.com/coq/coq/pull/11876

**Comment by jfehrle** [comment]
I haven't tried to verify whether deprecations in the doc were also implemented in the code.  It may not be fair to drop some items yet if they weren't in the code.

**Comment by Zimmi48** [comment]
Yes, `Include Type` cannot be removed without some changes first. See https://github.com/coq/coq/pull/11897 (BTW, I don't know when I will have time to work on this PR, someone taking over would be welcome).

Similarly, removing `Focus` / `Unfocus` is also blocked on implementing the request in #10688 first.

**Comment by SkySkimmer** [comment]
>Yes, Include Type cannot be removed without some changes first. See #11897 (BTW, I don't know when I will have time to work on this PR, someone taking over would be welcome).

I had a quick look but I don't really understand what's going on from the discussion.

**Comment by Zimmi48** [comment]
OK, I'll try to write a summary.

**Comment by ppedrot** [comment]
For Existential and Grab Existential Variables, we said that we should not be removing these commands for the time being because they can be used to work around bugs. Deprecation was just there to prevent people from using them carelessly.

**Comment by Zimmi48** [comment]
Hum, I thought that we had no examples leftover of such bugs? In any case, maybe waiting for one full release should help us determine if they really are still used in the wild...

**Comment by jfehrle** [comment]
@Zimmi48 or @gares can you confirm that ssr Search in the `ssrsearch` plugin (and the entire plugin) should be removed?

**Comment by Zimmi48** [comment]
It's deprecated since 8.12 and we've fixed in 8.12 or 8.13 all the issues which have been reported so far, so it's probably time. It would be easy to revert the removal if we get new issues reported that we cannot fix in time for the 8.14 release.

**Comment by CohenCyril** [comment]
> @Zimmi48 or @gares can you confirm that ssr Search in the `ssrsearch` plugin (and the entire plugin) should be removed?

I did not see this until I was notified by PR #13760.
So I will "backport" my comment: 

> Since 8.12 and 8.13 has not been extensively tested yet by mathcomp users, and that Search is one the most important features needed for mathcomp, I strongly advise not to remove the ssrsearch plugin for a few versions of Coq.

**Comment by jfehrle** [comment]
For removing `-extra-phony` and `-phony`, I see this in the code, which seems inconsistent

```
  | "-custom" :: _ ->
    error "Use \"-extra[-phony] target deps command\" instead of \"-custom command deps target\""
```

I also see these but it's not clear what should be removed.

```
  if project.makefile = None then
    eprintf "Warning: Omitting -o is deprecated\n\n";

  if project.subdirs <> [] then begin
    eprintf "Warning: Subdirectories are deprecated.\n";
    eprintf "Warning: Use double colon rules in %s.\n\n" local_file;
  end;
```

**Comment by gares** [comment]
I'm all in for removing deprecated stuff from coq_makefile, but I've no energy for that. I suggest making separate PRs to see what is still used (in spite of being deprecated) and what is not.

**Comment by gares** [comment]
> ` | "-custom" :: _ -> error "Use \"-extra[-phony] target deps command\" instead of \"-custom command deps target\""`

This is probably a message from coq_makefile1. In coq_makefil2 I think I did deprecate phony stuff, I don't recall custom, so I guess it is super-deprecated.

**Comment by gares** [comment]
>   if project.subdirs <> [] then begin

I don't recall which is the command line option for filling in subdirs, that one should go as well

**Comment by gares** [comment]
> ` if project.makefile = None then`

This is the only thing which worries me, in the sense that people may still be using this possibility.

**Comment by SkySkimmer** [comment]
After 
- https://github.com/coq/coq/pull/15194
- https://github.com/coq/coq/pull/15193

this should be everything.
Since neither is more important than the other and in the interest of not forgetting I'll just close this issue now.

**Comment by jfehrle** [comment]
Sure.  We can repeat the deprecation exercise sometime, but not me, not now.
